Remove "Uartmode : MIDI-38K" by adding /tmp/CORENAME

bbond007 opened the pull request:
Remove "Uartmode : MIDI-38K"

add creation of  /tmp/CORENAME to determine active core from within Linux. 

allows MidiLink.INI to have different [sections] per core, example:

[C64]
TCP_BAUD = 2400
[FXCAST]
TCP_BAUD = 19200
[AO486]
MIDI_BAUD = 38400

sorgelig commented:
then need to update Linux part as well. Otherwise with this update ao486 will stop to work.

sorgelig commented:
Let me know what parts i need to update on Linux side.

bbond007 commented:
Just /sbin/midilink needs to be updated

To clean things up, it might be a good idea to remove "4" option from the /sbin/uartmode scritpt but not necessary 

Thanks :)

sorgelig commented:
There are not many cores supporting UART, so it would be good to prepare midilink.ini file with presets for all supported cores. So users won't need to edit it.

bbond007 commented:
MIDI stuff for all cores should work correctly with default settings. MIDI speed ao486 is now assumed to be 38400 unless otherwise specified in the INI.  I did this to allows upgrade without requiring additional INI settings. 

The TCP (modem emulation) option does need tweaks for the C64 and Atari ST cores do to the fact that 115200 baud is just too fast for those cores. 

Here is my INI file. I can also PM it to you on Atari-Forum if you would like...

#This is the MidiLink INI File!
#Edit with Notepad++ under Windows

MIDILINK_PRIORITY   = -20
MUNT_OPTIONS        = 
MP3_VOLUME          = 100%
MUNT_VOLUME         = 100%
FSYNTH_VOLUME       = 100%
MIXER_CONTROL       = Master
FSYNTH_SOUNDFONT    = /media/fat/soundfonts/SC-55.sf2
UDP_SERVER          = 192.168.1.157
UDP_SERVER_PORT     = 1999
UDP_SERVER_FILTER   = FALSE
UDP_FLOW            = 0
TCP_SERVER_PORT     = 23
TCP_TERM_UPLOAD     = /media/fat/UPLOAD
TCP_TERM_DOWNLOAD   = /media/fat
TCP_TERM_MP3        = /media/fat/MP3
TCP_TERM_MIDI       = /media/fat/MIDI
TCP_TERM_SYNTH      = FluidSynth
TCP_TERM_ROWS       = 23
TCP_FLOW            = 0
TCP_SOUND           = TRUE
DELAYSYSEX          = TRUE

[MINIMIG]
TCP_TERM_ROWS       = 22

[C64]
TCP_BAUD            = 1200
MIDI_BAUD           = 2400
TCP_TERM_ROWS       = 24

[FXCAST]
TCP_BAUD            = 9600

[AO486]
MIDI_BAUD           = 38400

sorgelig commented:
here is enough.
thanks, i will update it.

sorgelig commented:
By the way, how about move MT32/SF folders to /media/fat/linux ?

bbond007 commented:
That is already easily done with FluidSynth as the soundfont is specified
in the INI:

FSYNTH_SOUNDFONT  = /media/fat/linux/soundfonts/SC-55.sf2

I will be adding a new option:

MUNT_ROM_PATH = /media/fat/linux/mt32-rom-data

Thanks

On Mon, Feb 25, 2019 at 1:34 PM sorgelig <notifications@github.com> wrote:

> By the way, how about move MT32/SF folders to /media/fat/linux ?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/MiSTer-devel/Main_MiSTer/pull/58#issuecomment-467126307>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACmkMcQVzaeMAkzWe0cAU7UkTS8pEdfSks5vRCy6gaJpZM4bOUMD>
> .
>

sorgelig commented:
let me know when you will be ready to include new midilink into release

bbond007 commented:
I think we are getting close. 
There were some bugs reported in the TCP modem emulation which I believe are now fixed fixed. 
I have also updated the default INI file. 
Thanks.

sorgelig commented:
So, it's ready for release or you still fix something?
I've read about some modem sounds you use. Do i need to add it to linux or it's separate addon?

bbond007 commented:
I think it is ready, but just to be sure I have asked the testers in the forum for feedback on the latest changes.

Nothing additional is needed for modem sounds. The default modem sounds are generated in code real-time  :) 

Users can optionally specify WAV files (in the INI) to use instead of the code-generated sounds, example:

TCP_SOUND_RING = /media/fat/linux/SOUNDS/ring.wav
TCP_SOUND_DIAL  = /media/fat/linux/SOUNDS/dial.wav
TCP_SOUND_CONNECT =/media/fat/linux/SOUNDS/connect.wav 
                                        
I think the default simulated sounds are pretty good, so I don't plan on including any WAV sounds.

Thanks.
